### ISSUE TITLE
Vary training implementation does not set attention_mask

### ISSUE THREAD

**Issue by dysby** [Ucas-HaoranWei/Vary#131]
Hi, thanks for sharing your research work!

I'm evaluating your training code base and seems to me that attention_mask is not properly set.

I followed for example the CaptionsDataset and the item output dictionary do not have the attention_mask key.

Wouldn't the attention_mask need to be set for all the prompt tokens (including the vision tokens), and the model should only compute the loss over the ground-truth response tokens? 

Best regards,

**Comment by dysby** [comment]
I was misunderstanding how training works and my comment is not correct.
